base/acme/src/org/dogtagpki/acme/validator/ACMEValidator.java

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme.validator;
+
+import javax.ws.rs.core.UriInfo;
+
+import org.dogtagpki.acme.ACMEAuthorization;
+import org.dogtagpki.acme.ACMEChallenge;
+
+/**
+ * @author Endi S. Dewata
+ */
+public class ACMEValidator {
```

Comment:
Yes, this can be an `abstract class`, not because it doesn't have any implementation, but because a base ACMEValidator class cannot really be used for anything useful. As discussed above, we will be moving common code from subclasses into the base class, so it cannot be an `interface`.